cpp/src/arrow/buffer-builder.h

```diff
@@ -164,18 +163,18 @@ class ARROW_EXPORT BufferBuilder {
 
   void Reset() {
     buffer_ = NULLPTR;
+    static_cast<void>(AllocateResizableBuffer(pool_, 0, &buffer_));
     capacity_ = size_ = 0;
   }
 
   int64_t capacity() const { return capacity_; }
   int64_t length() const { return size_; }
-  const uint8_t* data() const { return data_; }
-  uint8_t* mutable_data() { return data_; }
+  const uint8_t* data() const { return buffer_->data(); }
+  uint8_t* mutable_data() { return buffer_->mutable_data(); }
 
  private:
   std::shared_ptr<ResizableBuffer> buffer_;
   MemoryPool* pool_;
-  uint8_t* data_;
   int64_t capacity_;
```

Comment:
Just curious, if data_ can be replaced by buffer->data(), then capacity_ can also be replaced by buffer_->capacity() right?